src/App.tsx

```diff
@@ -8,37 +8,141 @@ import { PostsList } from './components/PostsList';
+          setComments(chosenPostComment);
+          setCommentsError(false);
+        })
+        .catch(() => setCommentsError(true))
+        .finally(() => setAreCommentsLoading(false));
+    }
+  }, [selectedPostId]);
+
+  const handleUserIdSelection = (id: number) => {
+    setSelectedUserId(id);
+  };
+
+  const handlePostIdSelection = (id: number) => {
+    setSelectedPostId(id);
+  };
+
+  const handleCommentAdd = (comment: CommentData) => {
+    const newComment = {
+      ...comment,
+      postId: selectedPostId,
+      id: Math.max(...comments.map(com => com.id)) + 1,
+    };
+
+    return addComment('/comments', newComment)
+      .then(() => setComments(prevComments => (
+        [...prevComments, newComment]
+      )));
+  };
+
+  const handleCommentDelete = (commentId: number) => {
+    setComments(comments.filter(comment => comment.id !== commentId));
+
+    const searchedUrl = `/comments/${commentId}`;
+
+    deleteComment(searchedUrl);
+  };
+
+  const chosenPost = posts.find(post => post.id === selectedPostId) || null;
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  onUserIdSelection={handleUserIdSelection}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUserId && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {arePostsLoading ? (
+                  <Loader />
+                ) : (
+                  <>
+                    {postsError && (
+                      <div
+                        className="notification is-danger"
+                        data-cy="PostsLoadingError"
+                      >
+                        Something went wrong!
+                      </div>
+                    )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                    {Boolean(posts.length) && (
+                      <PostsList
+                        posts={posts}
+                        onPostIdSelection={handlePostIdSelection}
+                      />
+                    )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                    {Boolean(!posts.length && selectedUserId) && (
```

Comment:
```suggestion
                    {!posts.length && selectedUserId && (
```
this already checks for boolean value

Reply:
There is the issue with 0 being displayed whenever the length  or  selectedUserId is equal to 0. That's why I converted it to Boolean value. I'm confused which expression would be more suitable here: posts.length === 0 && selectedUserId !== 0. This one seems to be a bit too long for me,  but on the other hand is more descriptive.

Reply:
second is more readable for me 👍